core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -149,6 +149,456 @@ public void testAliasAppearsInFuncArgs4() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("create table y ( x int, ts timestamp) timestamp(ts);");
+            ddl("create table y1 ( x int, ts timestamp) timestamp(ts);");
+            ddl("create table y2 ( x int, ts timestamp) timestamp(ts);");
+            final String query = "select  * from y \n" +
+                    "left join \n" +
+                    "y1 on \n" +
+                    "y1.x = y.x\n" +
+                    "INNER join (select LAST(ts) from y2) as y2 \n" +
+                    "on y2.LAST = y1.ts";
+            String queryNew = query + " union \n" + query;
+            final QueryModel model = compileModel(queryNew);
+            TestUtils.assertEquals("select-choose [y.x x, y.ts ts, y1.x x1, y1.ts ts1, y2.LAST LAST] y.x x, " +
+                    "y.ts ts, y1.x x1, y1.ts ts1, y2.LAST LAST from (select [x, ts] from y timestamp (ts) left join " +
+                    "select [x, ts] from y1 timestamp (ts) on y1.x = y.x join select [LAST] from (select-choose " +
+                    "[ts LAST] ts LAST from (select [ts] from y2 timestamp (ts)) order by LAST desc limit 1) y2 on " +
+                    "y2.LAST = y1.ts) union select-choose [y.x x, y.ts ts, y1.x x1, y1.ts ts1, y2.LAST LAST] y.x x," +
+                    " y.ts ts, y1.x x1, y1.ts ts1, y2.LAST LAST from (select [x, ts] from y timestamp (ts) " +
+                    "left join select [x, ts] from y1 timestamp (ts) on y1.x = y.x join select [LAST] from " +
+                    "(select-choose [ts LAST] ts LAST from (select [ts] from y2 timestamp (ts)) order by LAST desc " +
+                    "limit 1) y2 on y2.LAST = y1.ts)", model.toString0());
+            assertPlanNoLeakCheck(
+                    query,
+                    "SelectedRecord\n" +
+                            "    Hash Join Light\n" +
+                            "      condition: y2.LAST=y1.ts\n" +
+                            "        Hash Outer Join Light\n" +
+                            "          condition: y1.x=y.x\n" +
+                            "            DataFrame\n" +
+                            "                Row forward scan\n" +
+                            "                Frame forward scan on: y\n" +
+                            "            Hash\n" +
+                            "                DataFrame\n" +
+                            "                    Row forward scan\n" +
+                            "                    Frame forward scan on: y1\n" +
+                            "        Hash\n" +
+                            "            Sort light lo: 1\n" +
+                            "              keys: [LAST desc]\n" +
+                            "                SelectedRecord\n" +
+                            "                    DataFrame\n" +
+                            "                        Row forward scan\n" +
+                            "                        Frame forward scan on: y2\n");
```

Comment:
@puzpuzpuz shouldn't it be Backward Scan?
In query plan I can see limit and order by applied on designated timestamp column
**select-choose " +
                    "[ts LAST] ts LAST from (select [ts] from y2 timestamp (ts)) order by LAST desc limit 1**

Reply:
Yes, it should be a backward scan. Could you check why `SqlOptimizer` chooses Sort light lo: 1 instead of plain Limit lo: 1 on backward scan?

Reply:
@puzpuzpuz please check if this can be reason, I can further investigate then :
`model.getNestedModel().getJoinModels().getQuick(2).getNestedModel().getNestedModel()` 
which translates to `**select [ts] from y2 timestamp (ts)**`
is not getting populated with **orderByAdvice** due to which it is using `DataFrameFactory` which is having `followsOrderByAdvice`  as `false`, because of which it is using `LimitedSizeSortLightFactory` rather than `SelectRecordCursorFactory` 
This orderByAdvice should be populated in `rewriteOrderBy()` function of SqlOptimiser

Reply:
Yes, it could be the case that `optimiseOrderBy()` is misbehaving. Could you create a GH issue for this sub-optimal exec plan and leave a TODO with the link to the GH issue in this test? It's not necessary to fix the problem within this PR, so let's make sure to track it for now.

Reply:
No prob, up to you when to raise the issue. If you can come up with an example that doesn't rely on the rewrite from this PR and raise an issue, that would be great.